toolkit/tools/scheduler/schedulerutils/buildlist.go

```diff
@@ -107,22 +109,24 @@ func IsReservedFile(rpmPath string, reservedRPMs []string) bool {
 //   - packagesNamesToRebuild,
 //   - local packages listed in the image config, and
 //   - kernels in the image config (if built locally).
-func calculatePackagesToBuild(packagesNamesToBuild, packagesNamesToRebuild []*pkgjson.PackageVer, imageConfig, baseDirPath string, dependencyGraph *pkggraph.PkgGraph) (packageVersToBuild []*pkgjson.PackageVer, err error) {
+func calculatePackagesToBuild(packagesNamesToBuild, packagesNamesToRebuild []*pkgjson.PackageVer, imageConfig, baseDirPath string, dependencyGraph *pkggraph.PkgGraph, isBuildList bool) (packageVersToBuild []*pkgjson.PackageVer, err error) {
 	packageVersToBuild = append(packagesNamesToBuild, packagesNamesToRebuild...)
 
-	packageVersFromConfig, err := extractPackagesFromConfig(imageConfig, baseDirPath)
-	if err != nil {
-		err = fmt.Errorf("failed to extract packages from the image config, error:\n%w", err)
-		return
-	}
-
-	packageVersFromConfig, err = filterLocalPackagesOnly(packageVersFromConfig, dependencyGraph)
-	if err != nil {
-		err = fmt.Errorf("failed to filter local packages from the image config, error:\n%w", err)
-		return
+	packageVersFromConfig := []*pkgjson.PackageVer{}
+	err = nil
+	if isBuildList {
```

Comment:
I think we should move this logic inside `filterLocalPackagesOnly`. If tests are enabled and we end up building more packages because an image requires them, then we should also run their tests (assuming they have tests).